src/dom/cached-element-vector.ts

```diff
@@ -0,0 +1,157 @@
+/**
+ * CachedElementVector is used to optimize access to commonly checked values.
```

Comment:
Nice.  Thanks for adding comments.  Make its really clear and I think want to use this.